llvm/include/llvm/Frontend/OpenMP/ConstructDecompositionT.h

```diff
@@ -482,6 +484,28 @@ bool ConstructDecompositionT<C, H>::applyClause(
 
   return false;
 }
+// FIXME(JAN): Do the correct thing, but for now we'll do the same as collapse
+template <typename C, typename H>
+bool ConstructDecompositionT<C, H>::applyClause(
+    const tomp::clause::SizesT<TypeTy, IdTy, ExprTy> &clause,
+    const ClauseTy *node) {
+  // Apply "sizes" to the innermost directive. If it's not one that
+  // allows it flag an error.
+  if (!leafs.empty()) {
+    auto &last = leafs.back();
+
+    if (llvm::omp::isAllowedClauseForDirective(last.id, node->id, version)) {
+      last.clauses.push_back(node);
+      return true;
+    } else {
+      //      llvm::errs() << "** OVERRIDING isAllowedClauseForDirective **\n";
```

Comment:
@kparzysz I'm not sure if there is a way to override this in a cleaner way? Is it easy to introduce a new internal clause (e..g tile_sizes) and instead of just moving the sizes clause from tile to the outer construct, we instead translate it into the internal clause? I'm not sure if there are surrounding data structures that will make this hard, like location information and such?

Reply:
Yes, since we are moving the sizes clause to the parent directive from tile to whatever happens to enclose it. That's why I was thinking about adding a new internal clause to specify tiling (just like collapse is specified) in the case where the tiling is not stand alone. This will no longer be needed though once @Meinersbur's patch is merged and we would use a separate op for tiling instead.

Reply:
Instead of trying to pass information around, the parse tree is available when collecting the information to create the loop nest ops, so that's the solution I went with. We no longer try to move the sizes clause to the parent loop construct.